### DISCUSSION TITLE
How to reach a comment inside the pdf file?

### DISCUSSION THREAD

**Discussion by mawe1965** [UglyToad/PdfPig#343] [Q&A]
A pdf document can contain multiple comments. These are placed in a page and shown as yellow balloons. A comment can contain an attachment and these are shown as small paperclips.

Does anyone know if there is a object available in the library to access these objects?

Best regards,
Maarten Weers

**Comment by EliotJones** [comment]
Hi there, this sounds like annotations https://github.com/UglyToad/PdfPig#annotations-005

The embedded files will probably be accessible separately through the EmbeddedFiles API https://github.com/UglyToad/PdfPig#annotations-005

Let me know if this works :)

**Reply by mawe1965** [reply]
` Dim files As IReadOnlyList(Of EmbeddedFile) = Nothing

                If document.Advanced.TryGetEmbeddedFiles(files) AndAlso files.Count > 0 Then
                    For Each EmbeddedFile In files
                        Dim Filename As String = System.IO.Path.GetFileName(EmbeddedFile.FileSpecification)
                        Dim FileExtension As String = System.IO.Path.GetExtension(EmbeddedFile.FileSpecification)

... other code here

                    Next
                End If`

this only catches files stored in the attachment tab of the file, not the ones stored in a page.

**Comment by mawe1965** [comment]
Here the two different ways the files can be included in the PDF file:

![Untitled](https://user-images.githubusercontent.com/27855275/124090960-a168b000-da55-11eb-9006-4ce8c7377dbb.png)
![Untitled2](https://user-images.githubusercontent.com/27855275/124090975-a4fc3700-da55-11eb-8e38-977a9fe4cbf4.png)

**Reply by mawe1965** [reply]
I noticed I can find some info about the excel file from the annotation object. But how to get the excel file extracted is unknown to me.
![Untitled](https://user-images.githubusercontent.com/27855275/124094733-55b80580-da59-11eb-8395-76af3f65c347.png)

**Reply by EliotJones** [reply]
Are you able to share the file or an example file at all please? There are many many things in the PDF spec so it's hard to know exactly which feature this is without access to the raw file.

**Reply by mawe1965** [reply]
I am not able to share the files, they are all classified. If possible I will try to create one myself.
